app/pages/embed/list/[userId]/[[...pageNumber]].tsx

```diff
@@ -0,0 +1,177 @@
+    page = parseInt(pn[0]);
+    if (page.toString() !== pn[0] || page < 0) {
+      return { props: { error: 'Bad page number' } };
+    }
+  } else {
+    return { props: { error: 'Bad page number' } };
+  }
+
+  const [puzzles, hasNext] = await paginatedPuzzles(
+    page,
+    10,
+    'a',
+    params.userId
+  );
+
+  const props = {
+    props: {
+      userId: params.userId,
+      puzzles,
+      currentPage: page,
+      prevPage: page > 0 ? page - 1 : null,
+      nextPage: hasNext ? page + 1 : null,
+    },
+  };
+
+  const embedOptionsRes = await getCollection('em').doc(params.userId).get();
+  if (!embedOptionsRes.exists) {
+    return props;
+  }
+  const embedOptions = validate(embedOptionsRes.data());
+  if (!embedOptions) {
+    return props;
+  }
+  return { ...props, props: { ...props.props, embedOptions } };
+};
+
+export const getServerSideProps = withTranslation(gssp);
+
+export default function ThemedPage(props: PageProps | PageErrorProps) {
+  if ('error' in props) {
+    return (
+      <ErrorPage title="Error loading list">
+        <p>We&apos;re sorry, there was an error.</p>
+        <p>{props.error}</p>
+        <p>
+          Try the <Link href="/">homepage</Link>.
+        </p>
+      </ErrorPage>
+    );
+  }
+
+  let primary = PRIMARY;
+  let link = LINK;
+  let darkMode = false;
+  let preservePrimary = false;
+  if ('embedOptions' in props) {
+    primary = props.embedOptions?.p || PRIMARY;
+    link = props.embedOptions?.l || LINK;
+    darkMode = props.embedOptions?.d || false;
+    preservePrimary = props.embedOptions?.pp || false;
+    // Just ensure color is parseable, this'll throw if not:
+    parseToRgba(primary);
+  }
+
+  return (
+    <>
+      <Global
+        styles={{
+          body: {
+            backgroundColor: 'transparent !important',
+          },
+          'html, body.light-mode, body.dark-mode': colorTheme(
+            primary,
+            link,
+            darkMode,
+            preservePrimary
+          ),
+        }}
+      />
+      <EmbedContext.Provider value={true}>
+        <div css={{ margin: '1em' }}>
+          {props.puzzles.map((p, i) => (
+            <PuzzleResultLink
+              noTargetBlank={true}
+              fromEmbedPage={props.currentPage}
+              fullWidth={true}
+              key={i}
+              puzzle={p}
+              showDate={true}
+              showBlogPost={true}
+              showAuthor={false}
+              constructorIsPatron={false}
+              filterTags={[]}
+            />
+          ))}
+          {props.nextPage || props.prevPage !== null ? (
+            <p css={{ textAlign: 'center' }}>
+              {props.prevPage === 0 ? (
+                <Link
+                  css={{ marginRight: '2em' }}
+                  href={`/embed/list/${props.userId}`}
+                  noTargetBlank={true}
+                >
+                  ← Newer Puzzles
+                </Link>
+              ) : (
+                ''
```

Comment:
not-blocking, but thoughts on when to return `null` vs. `''` here? Seems like `null` could make it easier to style surrounding components, but maybe there's a benefit to `''`, or maybe they're equivalent?

Reply:
I'm honestly not sure why I started using the empty string - I'm wondering if there was an actual reason or just happenstance 😂